### PR TITLE
[SIDE-2283] Add development Webpack configuration

### PR DESCRIPTION
**Pull request by richseviora**
This PR:

- Adds a webpack development configuration that includes caching, no uglifying, and omits node_modules from builds. 
- Updates the readme with this, as well as moving parts of the readme around to better structure the content.

With caching, skipping babel on node_modules, minification off:
```
 SMP  ⏱  
General output time took 2.77 secs

 SMP  ⏱  Plugins
BannerPlugin took 0.001 secs

 SMP  ⏱  Loaders
babel-loader, and 
ts-loader took 1.6 secs
  module count = 1
modules with no loaders took 0.797 secs
  module count = 541
babel-loader took 0.263 secs
  module count = 15
```

### CODE REVIEW

File: ldk/javascript/src/webpack/config.ts [hunk 1]
```diff
@@ -1,96 +1,22 @@
 import * as webpack from 'webpack';
 import * as path from 'path';
-import Terser from 'terser-webpack-plugin';
-import { generateBanner } from './generate-banner';
-import { LdkSettings } from './ldk-settings'; // Need to dynamically refer to Loop's package.json
+import { LdkSettings } from './ldk-settings';
+import { buildBabelConfig, buildOptimization, buildWebpackConfig } from './shared';
 
+// Need to dynamically refer to Loop's package.json
```
Comment (RusGraf) [inline ldk/javascript/src/webpack/config.ts:6]: Rich, is this a TODO?

**Comment by RusGraf** [comment]
Do you think it will be helpful to update selfTestLoop to use: `config.development`?

**Review by RusGraf — APPROVED** [review]

**Review by cshields-oliveai — APPROVED** [review]

File: ldk/javascript/src/webpack/config.ts (hunk 1 above)
Comment (flowermm) [reply ldk/javascript/src/webpack/config.ts:6]: Looks like he is telling us why he disabled the linter, as he needs to use a dynamic require to get the loop's package.json

File: ldk/javascript/readme.md [hunk 2]
```diff
@@ -34,8 +34,50 @@ const merged = merge.merge(ldkConfig.default, {
...
+});
+
+module.exports = merged;
+```
+
+
+### Loading a Local Loop Into Olive Helps
+Once you have generated the Loop using the above steps into your `./dist/loop.js` directory, you can now load it into Olive Helps to test.
+
+- Open up Olive Helps and authenticate
```
Comment (flowermm) [inline ldk/javascript/readme.md:60]: Which wording do you use more frequently with your users? Login or Authenticate?

**Review by flowermm — APPROVED** [review]

File: ldk/javascript/src/webpack/config.ts (hunk 1 above)
Comment (richseviora) [reply ldk/javascript/src/webpack/config.ts:6]: What Michelle said :)

File: ldk/javascript/readme.md (hunk 2 above)
Comment (richseviora) [reply ldk/javascript/readme.md:60]: @flowermm  Hmm - good question. I would've opted for login or sign in myself. We haven't established a style guide of our own.

**Pull request merged**